### PR TITLE
TASK: Switch to Using Pipenv

### PR DESCRIPTION
**Pull request by mugizico**
TL;DR: like `bundler` but for Python.

Some of the benefit of using `Pipenv` vs the usual flow of `pip -> virtualenv -> requirements.txt` are:
 * no longer needing to  use pip and virtualenv separately. They work together.
* deterministic builds and dependency management ( directly related to issues we had with #77 , #78 )
* Pipenv uses Hashes to install/update dependencies which helps with discovering and preventing vulnerabilities in pypi packages.
* easily scannable and viewable  dependency graphy
```
pytest-cov==2.5.1
  - coverage [required: >=3.7.1, installed: 4.5.1]
  - pytest [required: >=2.6.0, installed: 3.6.3]
    - atomicwrites [required: >=1.0, installed: 1.1.5]
    - attrs [required: >=17.4.0, installed: 18.1.0]
    - more-itertools [required: >=4.0.0, installed: 4.2.0]
      - six [required: >=1.0.0,<2.0.0, installed: 1.11.0]
    - pluggy [required: >=0.5,<0.7, installed: 0.6.0]
    - py [required: >=1.5.0, installed: 1.5.4]
    - setuptools [required: Any, installed: 40.0.0]
    - six [required: >=1.10.0, installed: 1.11.0]
PyYAML==3.12
sqlalchemy-redshift==0.6.0
  - psycopg2 [required: >=2.5, installed: 2.7.3.2]
  - SQLAlchemy [required: >=0.9.2, installed: 1.1.9]
``` 
* `pipfile` and `pipfile.lock` instead of one `requirements.txt` file that is problematic to manage

I pushed a lambda tag successfully to S3 and will deploy it later

### CODE REVIEW

**Comment by MacHu-GWU** [comment]
`pipenv` would be the new community standard to manipulate virtualenv and dependencies. It is still new in the community and most of the developer may not be familiar with this, so I currently choose not using it now. It is still worth to explore deeper in it.

**Comment by lauraGgit** [comment]
@MacHu-GWU fwiw, it seems like the community is moving toward this.
